Character.cpp

```diff
@@ -87,31 +89,47 @@ int Character::takeDamage(int damage)
 }
 
 
-#include <assert>
+#include <assert.h>
 void Character::attackInternal(Character& other)
 {
     if( other.hitPoints <= 0 )
     {
         /*
-        When you defeat another Character: 
+        When you defeat another Character:
             a) your stats are restored to their initial value if they are lower than it.
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
-        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
+        
+        if (this->hitPoints < *this->initialHitPoints)
+            (this->hitPoints = *this->initialHitPoints);
+        this->hitPoints = this->hitPoints*1.1;
+        *this->initialHitPoints = this->hitPoints;
+        
+        if (this->armor < *this->initialArmorLevel)
+            (this->armor = *this->initialArmorLevel);
+        this->armor = this->armor*1.1;
+        *this->initialArmorLevel = this->armor;
+        
+        if (this->attackDamage < *this->initialAttackDamage)
+        (this->attackDamage = *this->initialAttackDamage);
+        this->attackDamage = this->attackDamage*1.1;
+        *this->initialArmorLevel = this->attackDamage;      
```

Comment:
You can D.R.Y. refactor this with a free function.   you are repeating the same logic 3x with 3 pairs of variables.
- [ ] revise.